EI-3331 - chore: Bump cryptography to 43.0.1

vtermanis opened the pull request:
Sanity performance results attached to ticket - looks OK.

vtermanis committed:
chore: EI-3331 - Bump cryptography to 43.0.1

vtermanis committed:
chore: EI-3331 - bump version to 2.1.2

- Bumping patch version despite slight incompatibilities between 42.0.8 &
  43.0.1 of cryptography since the SSL version mentioned therein was
  released a long time ago (03/2020)

vtermanis commented on setup.cfg:
```diff
@@ -18,7 +18,7 @@ max-complexity = 15
 [metadata]
 name = iotics-identity
-version = 2.1.1
+version = 2.1.2
```
See commit comment regarding reasoning around patch vs minor increment. Also:
- https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst#4300---2024-07-20
- https://mta.openssl.org/pipermail/openssl-announce/2020-March/000167.html

vtermanis committed:
chore: Fix README badges

- Switch to shields.io

vtermanis committed:
chore: Minor package metadata updates

- Switch email address
- Remove outdated Python version, add current ones (to classifiers)

vtermanis commented on README.md:
```diff
@@ -1,9 +1,8 @@
 # iotics-identity-py
 
-[![PyPi version](https://pypip.in/v/iotics-identity/badge.svg)](https://pypi.org/project/iotics-identity/)
```
These no longer worked

conbon commented:
any chance you could make a slight update to the readme instruction for `pip install .[dev]` to `pip install .'[dev]'`?

conbon reviewed (approved):
See comment - `zsh` shell problems on mac again. Common enough for users to hit. 
If the same works on linux it's worth updating I think.

vtermanis committed:
chore: Minor README.md command update

- Use single quotes for maximum compatibility with shells (e.g. zsh)
- Specify '-e' for development-mode install

vtermanis commented:
@conbon  c186e5d10
